text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,329 @@
+---
+rfc pr: [#347](https://github.com/aws/aws-cdk-rfcs/pull/347)
+tracking issue: https://github.com/aws/aws-cdk-rfcs/issues/66
+---
+
+# CDK Self Managed Stack Set Support
+
+As a CDK user, I would like to define stack set infrastructure through native CDK constructs,
+ and create/update self-managed stack sets through CDK deploy command.
+
+* **Original Author(s):** @linsona
+* **Tracking Issue**: #66
+* **API Bar Raiser**: @skinny85
+
+## Working Backwards
+
+### CHANGELOG
+
+* feat(assembly): add stack set artifact type and schema
+* feat(core): add `StackSet` construct
+* feat(cli): CLI commands like `deploy` now support stack sets
+* feat(bootstrap): add IAM Roles for performing CDK stack set deployments
+
+### README
+
+The `cdk.StackSet` construct in CDK defines AWS resources in a stack set and stack set
+ configuration/deployment preferences.
+ This is different from the `cdk.CfnStackSet` construct, which requires a CloudFormation stack that then deploys a CloudFormation stack set.
+ The `cdk.StackSet` construct creates the stack set directly if it does not exist,
+or updates a stack set and existing stack set instances while monitoring the stack set operation.
+
+#### Prerequisites
+
+To enable the use of self managed stack sets two roles must exist:
+
+* **Stack Set Administration Role:** Role in the parent account, which is used by CloudFormation to assume the stack set execution role in the child account.
+* **Stack Set Execution Role:** Role in the child accounts, which allows the stack set administration role in parent account to assume,
+and perform the stack set instance deployment.
```

Comment:
These roles sounds very much like the ones we use for CDK pipelines. Can't we reuse them?

Reply:
Yeah, so after discussing with Adam, we decided to go the route of introducing new roles:
* The `Stack Set Execution Role`  names should be identical in all accounts, which is a requirement for how stack sets operate. The current roles in the bootstrap template are suffixed with specific account ids
* The `Stack Set Execution Role` also must trust the `Stack Set Administration Role`, to assume and provision the stack set instance. 
  * In the current bootstrap, the trust flag would give any IAM entity in the account permissions to assume the execution role, which may have a large scope of permission. 
  * This is different from trust given to the ` DeploymentActionRole`, where it only has permissions to to create stack/changeset, and the actual role with large scope of permissions is `CloudFormationExecutionRole` which is not assumable to by the trusted account